### PR TITLE
add: support for `peft` in ddpo.

### PR DESCRIPTION
**Pull request by sayakpaul**
It's time.

Cc: @metric-space @younesbelkada

### CODE REVIEW

File: trl/models/modeling_sd_base.py [hunk 1]
```diff
@@ -583,42 +588,41 @@ def autocast(self):
...
-                    hidden_size=hidden_size, cross_attention_dim=cross_attention_dim
-                )
-            self.sd_pipeline.unet.set_attn_processor(lora_attn_procs)
-            return AttnProcsLayers(self.sd_pipeline.unet.attn_processors)
+            lora_config = LoraConfig(
+                r=4,
+                lora_alpha=4,
+                init_lora_weights="gaussian",
+                target_modules=["to_k", "to_q", "to_v", "to_out.0"],
+            )
```
Comment (sayakpaul) [inline trl/models/modeling_sd_base.py:605]: This matches with what was done previously.

File: trl/models/modeling_sd_base.py [hunk 2]
```diff
@@ -583,42 +592,41 @@ def autocast(self):
...
-                    block_id = int(name[len("up_blocks.")])
-                    hidden_size = list(reversed(self.sd_pipeline.unet.config.block_out_channels))[block_id]
-                elif name.startswith("down_blocks"):
-                    block_id = int(name[len("down_blocks.")])
-                    hidden_size = self.sd_pipeline.unet.config.block_out_channels[block_id]
-
-                lora_attn_procs[name] = LoRAAttnProcessor(
-                    hidden_size=hidden_size, cross_attention_dim=cross_attention_dim
-                )
-            self.sd_pipeline.unet.set_attn_processor(lora_attn_procs)
```
Comment (sayakpaul) [inline trl/models/modeling_sd_base.py:612]: No crazy layer iteration and state dict munging. Pretty please. Thanks to `peft`.

Comment (metric-space) [reply trl/models/modeling_sd_base.py:612]: Pretty swell that this is being cleanly replace. Sweet stuff

**Comment by metric-space** [comment]
hey @sayakpaul  things look splendid. Nothing has really changed in theory but would be nice to have a test run that shows convergence but I'll leave it to your and @younesbelkada's discretion to do without

**Review by metric-space — APPROVED** [review]

**Comment by sayakpaul** [comment]
@metric-space yes, will do! Thanks for your reviews.

**Comment by sayakpaul** [comment]
WandB run page: https://wandb.ai/sayakpaul/stable_diffusion_training/runs/7ebll3fb?workspace=user-sayakpaul.

**Comment by kashif** [comment]
LGTM! thanks

**Review by kashif — APPROVED** [review]

**Comment by younesbelkada** [comment]
Awesome work @sayakpaul and team !

**Pull request merged**